test/e2e/federation-replicaset.go

```diff
@@ -64,22 +71,129 @@ var _ = framework.KubeDescribe("Federation replicasets [Feature:Federation]", fu
 		})
 
 	})
+
+	// e2e cases for federated replicaset controller
+	Describe("Federated ReplicaSet", func() {
+		var (
+			clusters       map[string]*cluster
+			federationName string
+		)
+		BeforeEach(func() {
+			framework.SkipUnlessFederated(f.Client)
+			if federationName = os.Getenv("FEDERATION_NAME"); federationName == "" {
+				federationName = DefaultFederationName
+			}
+			registerClusters(clusters, UserAgentName, federationName, f)
```

Comment:
clusters is nil here.
This test is failing with:

```
Test Panicked
    assignment to entry in nil map
    /usr/local/go/src/runtime/panic.go:443

    Full Stack Trace
    	/usr/local/go/src/runtime/panic.go:443 +0x4e9
    k8s.io/kubernetes/test/e2e.registerClusters(0x0, 0x34c90e0, 0xe, 0x34c90d0, 0xa, 0xc820381040, 0x0, 0x0)
    	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/federation-util.go:207 +0xbc2
    k8s.io/kubernetes/test/e2e.glob.func32.2.1()
    	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/federation-replicaset.go:86 +0xf0
```

Need to add `clusters = map[string]*cluster{}`
Am sending a fix
cc @jianhuiz